src/pkg/systemartifact/dao/dao.go

```diff
@@ -0,0 +1,134 @@
+package dao
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/goharbor/harbor/src/lib/orm"
+	"github.com/goharbor/harbor/src/lib/q"
+	"github.com/goharbor/harbor/src/pkg/systemartifact/model"
+)
+
+const (
+	sizeQuery       = "select sum(size) as total_size from system_artifact"
+	totalSizeColumn = "total_size"
+)
+
+// DAO defines an data access interface for manging the CRUD and read of system
+// artifact tracking records
+type DAO interface {
+
+	// Create a system artifact tracking record.
+	Create(ctx context.Context, artifactRecord *model.SystemArtifact) (int64, error)
```

Comment:
artifactRecord is a little confusing here since we have artifact table.

Reply:
Renamed to `systemArtifact` to be consistent with the model naming conventions and avoid any confusions